### ISSUE TITLE
Lack of type-checking for function pointers

### ISSUE THREAD

**Issue by dangitall** [jonm/SillyMUD#48]
Code like this exists in `db.h` in the definition of `struct index_data`. There are a few more like it in in the `fight.c` (damage functions) and in the iteration functions in `hash.c`.

```
/* mob special procs are:
  int (*func)(struct char_data *, int, char *, char *, int);
  obj special procs are:
  int (*func)(struct char_data *, int, char *, struct obj_data *, int);
*/
int (*func) ();
```

**Comment by jonm** [comment]
It would appear this has been addressed at least in #60, #63, #64, #65, and #66. Is there anything remaining on this besides the hash table iterators in `hash.c` that you mentioned?

**Comment by dangitall** [comment]
I believe there are some outliers still:

```
protos.h:void room_iterate(struct room_data *rb[], void (*func) (), void *cdata);
protos.h:void hash_iterate(struct hash_header *ht, void (*func) (), void *cdata);
protos.h:struct char_data *find_mob_in_room_with_function(int room, int (*func) ());
```

ETA: (didn't mean to ignore that you already acked the funcitions in `hash.c`).
